spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestMetadataTableReadableMetrics.java

```diff
@@ -0,0 +1,498 @@
+            Maps.immutableEntry("binaryCol", "1111"),
+            Maps.immutableEntry("fixedCol", "1111"),
+            Maps.immutableEntry("dateCol", "1970-01-02"),
+            Maps.immutableEntry("timeCol", "00:00:00.000001"),
+            Maps.immutableEntry("timestampCol", "1970-01-01T00:00:00.000001")));
+
+    checkMetricValues(
+        row,
+        "upper_bound",
+        ImmutableMap.ofEntries(
+            Maps.immutableEntry("booleanCol", "true"),
+            Maps.immutableEntry("stringCol", "2"),
+            Maps.immutableEntry("intCol", "2"),
+            Maps.immutableEntry("longCol", "2"),
+            Maps.immutableEntry("floatCol", "0.0"),
+            Maps.immutableEntry("doubleCol", "2.0"),
+            Maps.immutableEntry("decimalCol", "2.00"),
+            Maps.immutableEntry("binaryCol", "2222"),
+            Maps.immutableEntry("fixedCol", "2222"),
+            Maps.immutableEntry("dateCol", "1970-01-03"),
+            Maps.immutableEntry("timeCol", "00:00:00.000002"),
+            Maps.immutableEntry("timestampCol", "1970-01-01T00:00:00.000002")));
+  }
+
+  @Test
+  public void testSelect() throws Exception {
+    String tableName = "testselect";
+    Table table = createTable(tableName, PRIMITIVE_SCHEMA, PartitionSpec.unpartitioned());
+
+    List<Record> records =
+        Lists.newArrayList(
+            createPrimitiveRecord(
+                false,
+                1,
+                1L,
+                0,
+                1.0D,
+                new BigDecimal("1.00"),
+                "1",
+                DateTimeUtil.dateFromDays(1),
+                DateTimeUtil.timeFromMicros(1),
+                DateTimeUtil.timestampFromMicros(1L),
+                Base64.getDecoder().decode("1111"),
+                ByteBuffer.wrap(Base64.getDecoder().decode("1111"))),
+            createPrimitiveRecord(
+                true,
+                2,
+                2L,
+                0,
+                2.0D,
+                new BigDecimal("2.00"),
+                "2",
+                DateTimeUtil.dateFromDays(2),
+                DateTimeUtil.timeFromMicros(2),
+                DateTimeUtil.timestampFromMicros(2L),
+                Base64.getDecoder().decode("2222"),
+                ByteBuffer.wrap(Base64.getDecoder().decode("2222"))));
+
+    DataFile dataFile =
+        FileHelpers.writeDataFile(table, Files.localOutput(temp.newFile()), records);
+    table.newAppend().appendFile(dataFile).commit();
+
+    Dataset<Row> nestedSelect =
+        spark
+            .read()
+            .format("iceberg")
+            .load("default." + tableName + ".files")
+            .select(
+                "readable_metrics.intCol.lower_bound", "readable_metrics.booleanCol.upper_bound");
+
+    List<Object[]> metricSelectResult = rowsToJava(nestedSelect.collectAsList());
+    assertEquals("Rows do not match", ImmutableList.of(row("1", "true")), metricSelectResult);
+
+    Dataset<Row> mixedSelect =
```

Comment:
Rather than this it may make sense to just use our "sql" method.

Could change this to 
```code
assertEquals("Failed nested select",
  expected,
  sql("SELECT ..."),
  )
  
```

We use this pattern in a bunch of places like

```
    assertEquals("Should return all expected rows", expected, sql("SELECT * FROM %s", tableName));
```

Reply:
Done, refactored all the tests to use this.